UT4MasterServer/Controllers/AccountController.cs

```diff
@@ -214,8 +216,23 @@ public IActionResult GetSSODomains()
 
 	[HttpPost("create/account")]
 	[AllowAnonymous]
-	public async Task<IActionResult> RegisterAccount([FromForm] string username, [FromForm] string email, [FromForm] string password)
+	public async Task<IActionResult> RegisterAccount([FromForm] string username, [FromForm] string email, [FromForm] string password, [FromForm] string recaptchaToken)
 	{
+		var reCaptchaSecret = this.configuration.GetValue<String>("ReCaptcha:SecretKey");
+		var httpClient = new HttpClient();
+		var httpResponse = httpClient.GetAsync($"https://www.google.com/recaptcha/api/siteverify?secret={reCaptchaSecret}&response={recaptchaToken}").Result;
+		if (httpResponse.StatusCode != System.Net.HttpStatusCode.OK)
+		{
+			return Conflict("Recaptcha validation failed");
+		}
+
+		var jsonResponse = httpResponse.Content.ReadAsStringAsync().Result;
+		dynamic jsonData = JObject.Parse(jsonResponse);
+		if (jsonData.success != true.ToString().ToLower())

```

Comment:
i'm unsure about exact syntax, but should be something like this:
```cs
if (!jsonData["success"].ToObject<bool>())
```

Reply:
I honestly copied and pasted this from someone's example, I tried to import a few nuget packages to handle it but neither worked. I am not attached to any of this code, change it as you see fit